Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -1454,4 +1454,10 @@ public boolean caresAboutManaColor() {
                 .filter(Objects::nonNull)
                 .anyMatch(Condition::caresAboutManaColor);
     }
+
+    public AbilityImpl copyWithZone(Zone zone) {
+        AbilityImpl copy = ((AbilityImpl)this.copy());
+        copy.zone = zone;
```

Comment:
miss new id, see old code

Reply:
Good catch, thanks! Fixed, and yeah I can't think of a use for copyWithZone that doesn't require a new id so best to put it here.